packages/callbacks/example/battery/lib/main.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:flutter/material.dart';
+import 'dart:ffi';
+import 'package:tizen_interop/6.5/tizen.dart';
+import 'package:tizen_interop_callbacks/tizen_interop_callbacks.dart';
+import 'package:tizen_log/tizen_log.dart';
+
+MyAppState? myAppState;
+
+void main() {
+  runApp(const MyApp());
+}
+
+class MyApp extends StatefulWidget {
+  const MyApp({super.key});
+
+  @override
+  State<MyApp> createState() => MyAppState();
+}
+
+class MyAppState extends State<MyApp> {
+  bool? _batteryCharging;
+  int? _batteryLevel;
+  final callbacks = TizenInteropCallbacks();
+
+  @override
+  void initState() {
+    super.initState();
+    myAppState = this;
+    initPlatformState();
+  }
+
+  void initPlatformState() {
+    final chargerCb =
+        callbacks.register<Void Function(Int32, Pointer<Void>, Pointer<Void>)>(
+            'device_changed_cb',
+            Pointer.fromFunction(_batteryChanged),
+            Pointer.fromAddress(0),
+            true);
+    int ret = tizen.device_add_callback(
+        device_callback_e.DEVICE_CALLBACK_BATTERY_CHARGING,
+        chargerCb.regCallbackPtr,
+        chargerCb.regUserData);
+
+    if (ret != 0) {
+      throw Exception('Failed to add battery charging callback');
+    }
+
+    final levelCb =
+        callbacks.register<Void Function(Int32, Pointer<Void>, Pointer<Void>)>(
+            'device_changed_cb', Pointer.fromFunction(_batteryChanged));
+    ret = tizen.device_add_callback(
+        device_callback_e.DEVICE_CALLBACK_BATTERY_CAPACITY,
+        levelCb.regCallbackPtr,
+        levelCb.regUserData);
+
+    // If you need to remove a callback, first unregister it:
+    // callbacks.unregister(chargerCb);
```

Comment:
I guess the user should clean the registered callback when not needed, and I'm afraid using dispose() for that would result in callback being unregistered too early - because user might not keep the reference to the object.

Reply:
You're providing this example to demonstrate best practices for using this package, and the example should be as "complete" as possible. It is natural that a callback registered in the `initState` method is unregistered in the `dispose` method of a state object (because they are counterpart of each other in the widget lifecycle). How could a callback be unregistered "too early"?